libraries/bezier-rs/src/subpath/solvers.rs

```diff
@@ -269,7 +269,33 @@ impl<ManipulatorGroupId: crate::Identifier> Subpath<ManipulatorGroupId> {
 
 	/// Does a path contain a point? Based on the non zero winding
 	pub fn contains_point(&self, target_point: DVec2) -> bool {
-		self.iter().map(|bezier| bezier.winding(target_point)).sum::<i32>() != 0
+		if self.closed() {
+			// Winding number for a closed curve.
+			return self.iter().map(|bezier| bezier.winding(target_point)).sum::<i32>() != 0;
+		}
+
+		// The curve is not closed. Pretend that it is and compute the winding number
+		// change as if we had a segment from the last point back to the first.
+
+		// Position of first and last anchor points.
+		let first_anchor_point = self.manipulator_groups.first().map(|group| group.anchor);
+		let last_anchor_point = self.manipulator_groups.last().map(|group| group.anchor);
+		let endpoints = first_anchor_point.zip(last_anchor_point);
+
+		// Weight interpolating intersection location from last to first anchor point. Reject weights outside of [0, 1].
+		let t = endpoints.map(|(first, last)| (target_point.y - last.y) / (first.y - last.y)).filter(|t| (0.0..=1.).contains(t));
+		// Compute point of intersection.
+		// Reject points that are right of the click location since we compute winding numbers by ray-casting left.
+		let intersection_point = endpoints.zip(t).map(|((first, last), t)| t * first + (1. - t) * last).filter(|p| p.x <= target_point.x);
+		let winding_modification = first_anchor_point.zip(intersection_point).map_or_else(
+			// None variant implies no intersection and no modification to winding number.
+			|| 0,
+			// Clockwise (decrement winding number) and counterclockwise (increment winding number) intersection respectively.
+			|(first, intersection)| if first.y >= intersection.y { -1 } else { 1 },
+		);
+
+		// Add the winding modification to the winding number of the rest of the curve.
+		self.iter().map(|bezier| bezier.winding(target_point)).sum::<i32>() + winding_modification != 0
```

Comment:
Couldn't this just involve temporarily closing the shape, calculating the value, then un-closing it?

Reply:
@Keavon this would require the function to take the mutable shape which is unnecessarily restrictive and would involve cloning in some cases. Although I agree that this could be simplified to use the existing winding function instead of re-implementing it:
```rs
let modification = endpoints.map_or(0, |(p1, p2)| bezier_rs::Bezier::from_linear_dvec2(p1, p2).winding(target_point));
```